### PR TITLE
Feature/add require on remove functions

### PR DESCRIPTION
**Pull request by Le-Caignec**
- add require on remove functions
- add units tests
- refactor priceOption to priceParam
- refactor subscriptionParams (remove useless subscriptionPrice)

### CODE REVIEW

File: packages/sharing-smart-contract/contracts/ProtectedDataSharing.sol [hunk 1]
```diff
@@ -298,15 +300,21 @@ contract ProtectedDataSharing is
         appForProtectedData[_collectionId][_protectedData] = _appAddress;
         protectedDataRegistry.safeTransferFrom(msg.sender, address(this), tokenId);
         protectedDatas[_collectionId][uint160(_protectedData)] = _protectedData;
+        protectedDataNumbersForCollection[_collectionId] += 1;
```
Comment (luisccastillo) [inline packages/sharing-smart-contract/contracts/ProtectedDataSharing.sol:303]: Shouldn't this counter be also modified when removing a protected data from collection?

Comment (Le-Caignec) [reply packages/sharing-smart-contract/contracts/ProtectedDataSharing.sol:303]: yeah my bad i forgot ! Thanks

Comment (Le-Caignec) [reply packages/sharing-smart-contract/contracts/ProtectedDataSharing.sol:303]: Fix

**Review by luisccastillo — APPROVED** [review]

File: packages/sharing-smart-contract/contracts/ProtectedDataSharing.sol [hunk 2]
```diff
@@ -41,6 +41,8 @@ contract ProtectedDataSharing is
     mapping(uint256 => mapping(uint160 => address)) public protectedDatas;
     // collectionId => (protectedDataAddress: address => App:address)
     mapping(uint256 => mapping(address => address)) public appForProtectedData;
+    // collectionId => protectedtedDataNumber
+    mapping(uint256 => uint256) public protectedDataNumbersForCollection;
```
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply packages/sharing-smart-contract/contracts/ProtectedDataSharing.sol]: Commit: 0aa7a24e4c939a9ce14693f7f8f8faebba085b9d

**Pull request merged**